crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -32,6 +32,73 @@ def role_constraints
     end
   end
 
+  def shutdown_openstack_at_nodes
+    db_nodes = []
+    NodeObject.find("state:crowbar_upgrade").each do |node|
```

Comment:
I think I had removing "crowbar_upgrade" there before and forgot to commit it...